tools/packaging/packaging.psm1

```diff
@@ -905,20 +905,29 @@ function Update-PSSignedBuildFolder
     $signedFilesFilter = Join-Path -Path $SignedFilesPathNormalized -ChildPath '*'
     Write-Verbose -Verbose "signedFilesFilter = $signedFilesFilter"
 
-    Get-ChildItem -Path $signedFilesFilter -Recurse -File | Select-Object -ExpandProperty FullName | ForEach-Object -Process {
-        Write-Verbose -Verbose "Processing $_"
+    $signedFilesList = Get-ChildItem -Path $signedFilesFilter -Recurse -File
+    foreach ($signedFileObject in $signedFilesList) {
+        # completely skip replacing pwsh on non-windows systems (there is no .exe extension here)
+        # and it may not be signed correctly
+        if ($signedFileObject.Name -eq "pwsh" -or $signedFileObject.Name -eq "Microsoft.PowerShell.GlobalTool.Shim.exe") {
+            Write-Verbose -Verbose "Skipping $signedFileObject"
+            continue
+        }
```

Comment:
```suggestion
        # completely skip replacing pwsh on non-windows systems (there is no .exe extension here)
        # and it may not be signed correctly
        
        # The Shim will no be signed in CI.
        
        if ($signedFileObject.Name -eq "pwsh" -or ($signedFileObject.Name -eq "Microsoft.PowerShell.GlobalTool.Shim.exe" -and $env:TF_BUILD)) {
            Write-Verbose -Verbose "Skipping $signedFileObject"
            continue
        }
```